test/testdata/noctx.go

```diff
@@ -0,0 +1,134 @@
+	cli.Do(req5)
+
+	req51, _ := newRequest(http.MethodPost, url, nil) // OK
+	req51 = req51.WithContext(ctx)
+	cli.Do(req51)
+
+	req52, _ := newRequestPkg(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+	cli.Do(req52)
+
+	type MyRequest = http.Request
+	f3 := func(req *MyRequest, ctx context.Context) *MyRequest {
+		return req
+	}
+	req6, _ := http.NewRequest(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+	req6 = f3(req6, ctx)
+
+	req61, _ := http.NewRequest(http.MethodPost, url, nil) // OK
+	req61 = req61.WithContext(ctx)
+	req61 = f3(req61, ctx)
+
+	type MyRequest2 http.Request
+	f4 := func(req *MyRequest2, ctx context.Context) *MyRequest2 {
+		return req
+	}
+	req7, _ := http.NewRequest(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+	req71 := MyRequest2(*req7)
+	f4(&req71, ctx)
+
+	req72, _ := http.NewRequest(http.MethodPost, url, nil) // OK
+	req72 = req72.WithContext(ctx)
+	req73 := MyRequest2(*req7)
+	f4(&req73, ctx)
+
+	req8, _ := func() (*http.Request, error) {
+		return http.NewRequest(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+	}()
+	cli.Do(req8)
+
+	req82, _ := func() (*http.Request, error) {
+		req82, _ := http.NewRequest(http.MethodPost, url, nil) // OK
+		req82 = req82.WithContext(ctx)
+		return req82, nil
+	}()
+	cli.Do(req82)
+
+	f5 := func(req, req2 *http.Request, ctx context.Context) (*http.Request, *http.Request) {
+		return req, req2
+	}
+	req9, _ := http.NewRequest(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+	req9, _ = f5(req9, req9, ctx)
+
+	req91, _ := http.NewRequest(http.MethodPost, url, nil) // OK
+	req91 = req91.WithContext(ctx)
+	req9, _ = f5(req91, req91, ctx)
+
+	req10, _ := http.NewRequest(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+	req11, _ := http.NewRequest(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+	req10, req11 = f5(req10, req11, ctx)
+
+	req101, _ := http.NewRequest(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+	req111, _ := http.NewRequest(http.MethodPost, url, nil) // OK
+	req111 = req111.WithContext(ctx)
+	req101, req111 = f5(req101, req111, ctx)
+
+	func() (*http.Request, *http.Request) {
+		req12, _ := http.NewRequest(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+		req13, _ := http.NewRequest(http.MethodPost, url, nil) // ERROR "should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext"
+		return req12, req13
+	}()
+
+	func() (*http.Request, *http.Request) {
```

Comment:
Does context have to be set in scope of the function where `Request` is created?

Reply:
If context is set it in other scope, it will be difficult to migrate to http.NewRequestWithContext.

WithContext should be run on the next line of NewRequest for compatibility of http.NewRequestWithContext function.